test/TestRist.cpp

```diff
@@ -0,0 +1,329 @@
+
+    std::condition_variable receiverCondition;
+    std::mutex receiverMutex;
+    receiver.networkDataCallback = [&](const uint8_t* buf, size_t size,
+                                       std::shared_ptr<RISTNetReceiver::NetworkConnection>& connection, rist_peer* peer,
+                                       uint16_t connectionId) {
+        EXPECT_EQ(connection, receiverCtx);
+        EXPECT_EQ(size, kBufferSize);
+        {
+            std::lock_guard<std::mutex> lock(receiverMutex);
+            EXPECT_EQ(*buf, nReceivedPackets + '0');
+            ++nReceivedPackets;
+        }
+        receiverCondition.notify_one();
+        return 0;
+    };
+    ASSERT_TRUE(receiver.initReceiver(receiverInterfaces, receiverSettings));
+
+    std::vector<std::tuple<std::string, int>> senderInterfaces{
+        std::tuple<std::string, int>("rist://127.0.0.1:8000", 0)};
+
+    RISTNetSender::RISTNetSenderSettings senderSettings;
+    senderSettings.mPSK = kValidPsk;
+    RISTNetSender sender;
+    ASSERT_TRUE(sender.initSender(senderInterfaces, senderSettings));
+
+    std::vector<uint8_t> sendBuffer(kBufferSize);
+    for (auto i = 0; i < kSentPackets; i++) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(50));
+        std::fill(sendBuffer.begin(), sendBuffer.end(), i + '0');
+        EXPECT_TRUE(sender.sendData((const uint8_t*)sendBuffer.data(), sendBuffer.size()));
+    }
+
+    {
+        std::unique_lock<std::mutex> lock(receiverMutex);
+        bool successfulWait = receiverCondition.wait_for(lock, std::chrono::seconds(2),
+                                                         [&]() { return nReceivedPackets == kSentPackets; });
+        EXPECT_TRUE(successfulWait) << "Timeout waiting for receiving data from sender, sent " << kSentPackets
+                                    << " packet(s) and received " << nReceivedPackets << " packet(s).";
+    }
+}
+
+// TODO Enable test when STAR-38 is fixed.
+TEST(TestRist, DISABLED_TestPsk) {
+    RISTNetReceiver receiver;
+    std::vector<std::string> receiverInterfaces{"rist://@0.0.0.0:8000"};
+    RISTNetReceiver::RISTNetReceiverSettings receiverSettings;
+    receiverSettings.mPSK = kValidPsk;
+
+    receiver.validateConnectionCallback = [&](const std::string& ipAddress, uint16_t port) {
+        return std::make_shared<RISTNetReceiver::NetworkConnection>();
+    };
+
+    ASSERT_TRUE(receiver.initReceiver(receiverInterfaces, receiverSettings));
+
+    std::vector<std::tuple<std::string, int>> senderInterfaces{
+        std::tuple<std::string, int>("rist://127.0.0.1:8000", 0)};
+    RISTNetSender::RISTNetSenderSettings senderSettings;
+    senderSettings.mPSK = kInvalidPsk;
+    RISTNetSender sender;
+    EXPECT_FALSE(sender.initSender(senderInterfaces, senderSettings));
+
+    senderSettings.mPSK = kValidPsk;
+    EXPECT_TRUE(sender.initSender(senderInterfaces, senderSettings));
+}
+
```

Comment:
I miss a test case for the OOB (out-of-band) communication. That should be possible to use for sending metadata between the sender and the receiver. I think it should be possible to send in both directions, but the "back" direction (receiver to sender) is most interesting.

Reply:
Right, I forgot about that!